src/partition/repart.c

```diff
@@ -6376,6 +6378,9 @@ static int help(void) {
                "     --offline=BOOL       Whether to build the image offline\n"
                "  -s --copy-source=PATH   Specify the primary source tree to copy files from\n"
                "     --copy-from=IMAGE    Copy partitions from the given image(s)\n"
+               "  -S --make-ddi=sysext    Make a system extension DDI\n"
```

Comment:
Not entirely sure if I like the name --make-ddi=. Technically systemd-repart always makes a DDI.

Reply:
not really, the original usecase was to grow partitions on first boot. Thus it didn't "make" anything, it just "grew" something, or "added" it.

Any strictly speaking it is not bound to the Discoverable Partition Specification, you can make arbitrary partitions with it, for example "Generic Linux" partitions too if you like. "DDI" I would say refers to disk images implementing the d-p-s...

But this new switch it strictly *builds* images, it cannot operate on existing ones, and it strictly generates images that implement the DPS, because the three templates this PR provides impement the dps.

Reply:
But I am not bound to the name at all! Happy to switch to a better name!